packages/synapse-react-client/src/components/EntityLink.tsx

```diff
@@ -64,16 +64,19 @@ export const EntityLink = (props: EntityLinkProps) => {
     } else {
       type = getEntityTypeFromHeader(entity)
     }
-
     if (link) {
       return (
         <Link
           className={className}
           target="_blank"
           rel="noopener noreferrer"
-          href={`${PRODUCTION_ENDPOINT_CONFIG.PORTAL}#!Synapse:${entity.id!}${
-            versionNumber ? `.${versionNumber}` : ''
-          }`}
+          href={
+            typeof link === 'string'
+              ? link
+              : `${PRODUCTION_ENDPOINT_CONFIG.PORTAL}#!Synapse:${entity.id!}${
+                  versionNumber ? `.${versionNumber}` : ''
+                }`
```

Comment:
This error was here before your change, but can you change `PRODUCTION_ENDPOINT_CONFIG.PORTAL` to `getEndpoint(BackendDestinationEnum.PORTAL_ENDPOINT)`?

This ensures that the link keeps you on the stack that you were already on, e.g. if you click this in staging, it should keep you in staging. Right now, it always takes you to production, which is wrong!